tests/unit/s2n_server_hello_retry_test.c

```diff
@@ -91,32 +84,26 @@ int main(int argc, char **argv)
             + COMPRESSION_METHOD_SIZE;
 
         server_conn->actual_protocol_version = S2N_TLS13;
+        server_conn->server_protocol_version = S2N_TLS13;
+        server_conn->client_protocol_version = S2N_TLS13;
         server_conn->secure.cipher_suite = &s2n_ecdhe_ecdsa_with_aes_128_gcm_sha256;
         server_conn->secure.server_ecc_evp_params.negotiated_curve = s2n_all_supported_curves_list[0];
         server_conn->secure.client_ecc_evp_params[0].negotiated_curve = s2n_all_supported_curves_list[0];
         EXPECT_SUCCESS(s2n_ecc_evp_generate_ephemeral_key(&server_conn->secure.client_ecc_evp_params[0]));
 
-        s2n_set_connection_hello_retry_flags(server_conn);
+        EXPECT_SUCCESS(s2n_set_connection_hello_retry_flags(server_conn, 1));
 
         /* The client will need a key share extension to properly parse the hello */
         /* Total extension size + size of each extension */
         total += 2 + s2n_extensions_server_supported_versions_size(server_conn)
                 + s2n_extensions_server_key_share_send_size(server_conn);
 
-        EXPECT_EQUAL(s2n_is_hello_retry_required(server_conn), 1);
+        EXPECT_TRUE(s2n_is_hello_retry_message(server_conn));
         EXPECT_SUCCESS(s2n_server_hello_retry_send(server_conn));
 
         EXPECT_EQUAL(s2n_stuffer_data_available(server_stuffer), total);
```

Comment:
This just verifies the right amount of data was written, and the _size methods are deprecated anyway. Is there a better way to verify that the retry message was processed?

Reply:
This is the test for only `server_hello_retry_send`, `server_hello_retry_recv` tests are within `s2n_client_hello_retry_test` file.